### PR TITLE
[Banner] fix image constraint when it is hidden and on single row mode.

### PR DESCRIPTION
**Pull request by fluxminds**
After adding support for single row mode on Banner, image constraint needs to be updated. Otherwise it will break the vertical constraints because of the height change.

### CODE REVIEW

File: components/Banner/src/MDCBannerView.m [hunk 1]
```diff
@@ -48,9 +48,9 @@ @interface MDCBannerView ()
 
 // Image constraints
 @property(nonatomic, readwrite, strong) NSLayoutConstraint *imageViewConstraintLeading;
-@property(nonatomic, readwrite, strong) NSLayoutConstraint *imageViewConstraintTopSmall;
+@property(nonatomic, readwrite, strong) NSLayoutConstraint *imageViewConstraintCenterY;
```
Comment (codeman7) [inline components/Banner/src/MDCBannerView.m:51]: Please add docs.

File: components/Banner/src/MDCBannerView.m [hunk 2]
```diff
@@ -195,15 +195,11 @@ - (void)setUpImageViewConstraints {
...
-      [self.imageView.topAnchor constraintEqualToAnchor:self.layoutMarginsGuide.topAnchor
-                                               constant:kTopPaddingSmall];
   self.imageViewConstraintTopLarge =
       [self.imageView.topAnchor constraintEqualToAnchor:self.layoutMarginsGuide.topAnchor
                                                constant:kTopPaddingLarge];
-  self.imageViewConstraintBottom =
-      [self.imageView.bottomAnchor constraintEqualToAnchor:self.layoutMarginsGuide.bottomAnchor
-                                                  constant:-kBottomPadding];
+  self.imageViewConstraintCenterY =
+      [self.imageView.centerYAnchor constraintEqualToAnchor:self.buttonContainerView.centerYAnchor];
```
Comment (codeman7) [inline components/Banner/src/MDCBannerView.m:202]: Should this be centered to self.centerYAnchor?

**Review by codeman7 — APPROVED** [review]
Please add screenshot to the commit message or snapshot test to capture the improvement so we don't regress later.

File: components/Banner/src/MDCBannerView.m (hunk 2 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.m:202]: Button, text and button are not centered to the view on Y axis according to the guideline. There is a one point off.

File: components/Banner/src/MDCBannerView.m (hunk 1 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.m:51]: Followed up in https://github.com/material-components/material-components-ios/issues/7650. Will be done altogether when finalized.

**Comment by fluxminds** [comment]
Snapshot tests have been added.

**Pull request merged**